### PR TITLE
libx264: get it building using mingw on Windows

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libx264/all**

Get it building on mingw on Windows

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libx264/all/conanfile.py [hunk 1]
```diff
@@ -40,80 +50,94 @@ def configure(self):
...
+        if self.settings.build_type == "Debug":
+            args.append("--enable-debug")
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            # bitstream-a.S:29:18: error: unknown token in expression
+            args.append("--extra-asflags=-arch arm64")
+            args.append("--extra-ldflags=-arch arm64")
+
+        if self._with_nasm:
+            # FIXME: get using user_build_info
+            self._override_env["AS"] = os.path.join(self.deps_cpp_info["nasm"].rootpath, "bin", "nasm{}".format(".exe" if tools.os_info.is_windows else "")).replace("\\", "/")
```
Comment (SpaceIm) [inline recipes/libx264/all/conanfile.py:102]: Is it the fix for MinGW?

Comment (madebr) [reply recipes/libx264/all/conanfile.py:102]: The error was:
```
----Running------
> C:\.conan\36de63\1\bin\msys64\usr\bin\bash.exe --login -c ^"cd \^"/c/users/maarten/.conan/data/libx264/20191217/_/_/build/526e61ab394798c6e72c93aafa8d2122e49dc8fd\^" ^&^& PATH=\^"/c/users/maarten/.conan/data/mingw-w64/8.1/_/_/package/6903a9d1b48b06f7fbe0929db654512c77e6cc32/bin:/c/users/maarten/.conan/data/nasm/2.15.05/_/_/package/0a420ff5c47119e668867cdb51baff0eca1fdb68/bin:/c/.conan/36de63/1/bin/msys64/usr/bin:$PATH\^" ^&^& /c/users/maarten/.conan/data/libx264/20191217/_/_/build/526e61ab394798c6e72c93aafa8d2122e49dc8fd/source_subfolder/configure --disable-cli --prefix=/c/users/maarten/.conan/data/libx264/20191217/_/_/package/526e61ab394798c6e72c93aafa8d2122e49dc8fd --enable-static --bit-depth=all  ^"
-----------------
Found GNU assembler (GNU Binutils) 2.30
Minimum version is nasm-2.13
If you really want to compile without asm, configure with --disable-asm.
```
I noticed the version of nasm added as a build requirement was sufficient recent.
So the configure script must have failed to detect it.

**Review by SpaceIm — APPROVED** [review]

**Review by SSE4 — COMMENTED** [review]
many changes in recipe, are all of them actually needed to build for MinGW?

**Comment by madebr** [comment]
> 
> 
> many changes in recipe, are all of them actually needed to build for MinGW?

No, only the nasm change.
The rest is "modernization".

**Review by SSE4 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**